### DISCUSSION TITLE
How can I change the error code for exceptions so that it's not 200 OK?

### DISCUSSION THREAD

**Discussion by ssona09** [Netflix/dgs-framework#1039] [Q&A]
I want to change the error code from 200 to 400 for instances where an invalid query is requested. I know I can see the errors field in the response but I want to change the status code. How can I achieve this?

**Comment by srinivasankavitha** [comment]
You can implement a custom exception handler and update the errors in the
response:
https://netflix.github.io/dgs/error-handling/#mapping-custom-exceptions

**Reply by ssona09** [reply]
So I can’t change the error status to code to something other than 200? I can just change the error type and message?

**Reply by paulbakker** [reply]
In GraphQL there can be a mix of successful data and errors - so changing the status code isn't the right thing to do.
If you would want to hook into this you would currently need to implement your own [RestController](https://github.com/Netflix/dgs-framework/blob/master/graphql-dgs-spring-webmvc/src/main/kotlin/com/netflix/graphql/dgs/mvc/DgsRestController.kt). 
I really don't recommend doing that though :-)

**Reply by shadikkhan** [reply]
Atleast there should be some ways to modify HTTP status code in this case. Updating in error object is not enough in term where things are relying on http status code

**Comment by paulbakker** [comment]
> where an invalid query is requested

If the query is incorrect (e.g. incorrect syntax) it should be 400 already. Which is implemented here: https://github.com/Netflix/dgs-framework/blob/master/graphql-dgs-spring-webmvc/src/main/kotlin/com/netflix/graphql/dgs/mvc/DgsRestController.kt#L122

However, for a query where there are just some fields that have errors, it would be against the GraphQL spec to fail the request completely, because you can have a mix of successful and error fields.
If you want to do that anyway, you could write a filter.
